src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/commands.py

```diff
@@ -6,17 +6,84 @@
+        client_factory=cf_hdinsight_applications
+    )
+
+    hdinsight_extensions_sdk = CliCommandType(
+        operations_tmpl='azure.mgmt.hdinsight.operations.extensions_operations#ExtensionsOperations.{}',
+        client_factory=cf_hdinsight_extensions
+    )
+
+    hdinsight_locations_sdk = CliCommandType(
+        operations_tmpl='azure.mgmt.hdinsight.operations.locations_operations#LocationsOperations.{}',
+        client_factory=cf_hdinsight_locations
+    )
+
+    hdinsight_script_execution_history_sdk = CliCommandType(
+        operations_tmpl='azure.mgmt.hdinsight.operations.script_execution_history_operations#'
+                        'ScriptExecutionHistoryOperations.{}',
+        client_factory=cf_hdinsight_script_execution_history
+    )
+
+    # cluster operations
     with self.command_group('hdinsight', hdinsight_clusters_sdk, client_factory=cf_hdinsight_clusters) as g:
         g.custom_command('create', 'create_cluster', supports_no_wait=True)
         g.command('resize', 'resize', supports_no_wait=True)
         g.show_command('show', 'get')
         g.custom_command('list', 'list_clusters')
         g.wait_command('wait')
         g.command('delete', 'delete', confirmation=True, supports_no_wait=True)
+
+    # usage operations
+    with self.command_group('hdinsight', hdinsight_locations_sdk, client_factory=cf_hdinsight_locations) as g:
+        g.command('usage', 'list_usages')
+
+    # script action operations
+    with self.command_group('hdinsight script-action',
+                            hdinsight_script_actions_sdk,
+                            client_factory=cf_hdinsight_script_actions) as g:
+        g.show_command('show', 'get_execution_detail')
+        g.custom_command('list', 'list_hdi_script_actions')
+        g.command('delete', 'delete')
+        g.custom_command('execute',
+                         'execute_hdi_script_action',
+                         command_type=hdinsight_clusters_sdk,
+                         client_factory=cf_hdinsight_clusters)
+        g.command('promote',
+                  'promote',
+                  command_type=hdinsight_script_execution_history_sdk,
+                  client_factory=cf_hdinsight_script_execution_history)
+
+    # application operations
+    with self.command_group('hdinsight application',
+                            hdinsight_applications_sdk, client_factory=cf_hdinsight_applications) as g:
+        g.custom_command('create', 'create_hdi_application')
+        g.show_command('show', 'get')
+        g.command('list', 'list_by_cluster')
+        g.wait_command('wait')
+        g.command('delete', 'delete', confirmation=True, supports_no_wait=True)
+
+    # OMS operations
+    with self.command_group('hdinsight oms', hdinsight_extensions_sdk, client_factory=cf_hdinsight_extensions) as g:
+        g.show_command('show', 'get_monitoring_status')
+        g.custom_command('enable', 'enable_hdi_monitoring')
+        g.command('disable', 'disable_monitoring')
```

Comment:
combine enable and disable into a update command

Reply:
@williexu Could you share some thoughts about advantages of the change?

There are 2 reasons we want to use `enable` and `disable` directly instead of `update`:
1. From customer's perspective, he/she has to use `az hdinsight oms update --enable`, `az hdinsight oms update --disable`. It is more complex than `az hdinsight oms enable` and `az hdinsight oms disable`
2. From our side, considering there are 2 methods `enable_monitoring` and `disable_monitoring` in SDK, we just want to keep aligned.

Reply:
The change will be consistent with the rest of the CLI commands. You will like just need one `--enable true/false` flag.
A CLI customer will not know the methods in the sdk, but they will be using other CLI commands.

Reply:
@williexu , I think `update --enable true/false` is even more complicated. Obviously `enable/disable` is more brief. For OMS, there is nothing else to `update`, so a `update` command seems too heavy.

Besides, I searched the existing commands and find there are some `enable/disable` commands in the repo:
- `vm boot-diagnostics enable/disable`
- `vm encryption enable/disable`
- `vmss encryption enable/disable`
- `batch job enable/disable`
- `batch job-schedule enable/disable`
- `batch node scheduling enable/disable`
- `advisor recommendation enable/disable`

Reply:
@idear1203 the examples you found are exceptions to the vast majority of the commands in the CLI. However, since youre sure, that there will not be any update functionality, we can leave the commands as is.